docs/architecture/README.md

```diff
@@ -60,6 +60,9 @@ This document specifies the architecture for GCP Ingestion as a whole.
    - Must use a 5XX error error code
 - Must accept configuration mapping `uri` to PubSub Topic
    - Expected initial topics are Structured Ingestion, Telemetry, and Pioneer
+- Must accept configuration defining HTTP headers to capture
+   - Must store configured headers as PubSub message attributes
+   - Must name attributes as the header name in lowercase with `-` converted to `_`
```

Comment:
Does this bit cover what you think needs to be communicated about headers, @whd?

Reply:
I'd put this in the `METADATA_HEADERS` section of https://github.com/mozilla/gcp-ingestion/blob/master/ingestion-edge/README.md#configuration since it's an implementation detail and not really high-level architecture.  Though this does raise the point that the high-level architecture doc makes no statements on the configuration or ability of the edge to preserve headers, and the [edge specification](https://github.com/mozilla/gcp-ingestion/blob/master/docs/edge.md#edge-server-pubsub-message-schema) also mentions headers only in passing in the schema definition, and that notably `x-pipeline-proxy` is not present there.

With apologies for not doing it myself; as I stated, I think for the purpose for which this discussion arose simply conveying the explicit header name that the client should supply and the expected value of the header from this documentation is sufficient. I am actively trying not to bikeshed too much on documentation which is why I approved this as-is; for my purposes the original PR suffices.